engine/flink/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/source/flink/FlinkKafkaSchemaRegistryBasedKeyValueDeserializationSchemaFactory.scala

```diff
@@ -0,0 +1,82 @@
+package pl.touk.nussknacker.engine.schemedkafka.source.flink
+
+import io.confluent.kafka.schemaregistry.ParsedSchema
+import org.apache.flink.api.common.typeinfo.{TypeInformation, Types}
+import org.apache.flink.api.java.typeutils.ResultTypeQueryable
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.serialization.Deserializer
+import pl.touk.nussknacker.engine.flink.api.typeinformation.TypeInformationDetection
+import pl.touk.nussknacker.engine.kafka.KafkaConfig
+import pl.touk.nussknacker.engine.kafka.consumerrecord.ConsumerRecordKafkaDeserializationSchema
+import pl.touk.nussknacker.engine.kafka.serialization.KafkaDeserializationSchema
+import pl.touk.nussknacker.engine.schemedkafka.RuntimeSchemaData
+import pl.touk.nussknacker.engine.schemedkafka.flink.typeinfo.ConsumerRecordTypeInfo
+import pl.touk.nussknacker.engine.schemedkafka.schemaregistry.SchemaRegistryClientFactory
+import pl.touk.nussknacker.engine.schemedkafka.schemaregistry.serialization.{
+  AbstractSchemaRegistryBasedDeserializerFactory,
+  SchemaRegistryBasedDeserializerFactory
+}
+import pl.touk.nussknacker.engine.schemedkafka.serialization.KafkaSchemaBasedKeyValueDeserializationSchemaFactory
+
+import scala.reflect.ClassTag
+
+class FlinkKafkaSchemaRegistryBasedKeyValueDeserializationSchemaFactory(
```

Comment:
I think that we can do a little bit more to reduce or totally remove the code duplication.
First of all we can inherit from `KafkaSchemaBasedKeyValueDeserializationSchemaFactory`
Secondly, we can use the adapter pattern to enrich the returned deserialization schema with the the produced type info